uaclient/entitlements/repo.py

```diff
@@ -72,6 +76,28 @@ def _check_for_reboot(self) -> bool:
     def repo_key_file(self) -> str:
         pass
 
+    def can_disable(
+        self, ignore_dependent_services: bool = False
+    ) -> Tuple[bool, Optional[CanDisableFailure]]:
+        result, reason = super().can_disable(
+            ignore_dependent_services=ignore_dependent_services
+        )
+        if result is False:
+            return result, reason
+
+        if not self.origin and self.purge:
+            return (
+                False,
+                CanDisableFailure(
+                    CanDisableFailureReason.NO_PURGE_WITHOUT_ORIGIN,
+                    messages.E_REPO_PURGE_FAIL_NO_ORIGIN.format(
+                        entitlement_name=self.title, title=self.title
+                    ),
+                ),
+            )
```

Comment:
This is an error message for a bug if we, the developers, introduce a repo-entitlement without an origin. I wish we could structure the code in such a way that this is so unlikely as to not deserve a user-facing error message. But this is probably the right thing to do, I'm just complaining